esp_pixel_driver.ino

```diff
@@ -1,135 +1,165 @@
 
-//Calculated Constants
-const int channels = STRIP_LENGTH * 3;
-const int maxUniverses = channels / 512 + ((channels % 512) ? 1 : 0);
+// Calculated Constants
+int channels;
+int maxUniverses;
 
-//DMX Handler Globals
+// DMX Handler Globals
 ArtnetWifi artnet;
-bool universesReceived[maxUniverses];
-bool sendFrame = 1;
+bool *universesReceived;
 int previousDataLength = 0;
+bool dmxReceived;
 
-//Network Config
-const IPAddress ip(192, 168, 1, 200);
+// Network Config
+const IPAddress ip(192, 168, 1, 249);
 const IPAddress gateway(192, 168, 1, 1);
 const IPAddress subnet(255, 255, 255, 0);
 
-void setup()
-{
-  //LED Init
-  FastLED.addLeds<WS2812, OUTPUT_PIN, RGB>(leds, STRIP_LENGTH);
+QueueHandle_t dmxQueue = xQueueCreate(1, sizeof(DmxParameters));
 
-  //Serial Start
+void setup() {
+
+  // Serial Start
   Serial.begin(115200);
   delay(500);
 
-  //Status - Connecting to Wifi
+  if (dmxQueue == NULL) {
+    Serial.println("Error creating the queue");
+  }
+
+  xTaskCreatePinnedToCore(dmxTask,          /* Function to implement the task */
+                          "DMX Frame Task", /* Name of the task */
+                          100000,           /* Stack size in words */
+                          NULL,             /* Task input parameter */
+                          100,              /* Priority of the task */
+                          NULL,             /* Task handle. */
+                          1);               /* Core where the task should run */
+
+  // Load Config from memory, apply static overrides, and run wm setup
+  init_config();
+
+  // LED Init
+  leds = (CRGB *)malloc(sizeof(CRGB) * config.strip_length);
+  if (!leds) { // failed malloc returns null
+    Serial.println(
+        "Failed to allocate memory for LED Array. Reduce Strip Length.");
+    Serial.println("Returning to Configuration Portal...");
+    WifiManagerAdapter::setup(&config);
+  }
+  FastLED.addLeds<WS2812, OUTPUT_PIN, RGB>(leds, config.strip_length);
+
+  // Status - Connecting to Wifi
   status(0);
   setup_wifi();
   artnet.begin();
   delay(500);
 
-  //Status - Waiting for Artnet
+  pixel_server.begin();
+  delay(500);
+
+  // Status - Waiting for Artnet
   status(1);
   artnet.setArtDmxCallback(onDmxFrame);
   delay(500);
 }
 
-void loop()
-{
+void loop() {
+  // handle web requests
+  pixel_server.Listen();
+  // listen for artnet
   artnet.read();
-  delay(5);
 }
 
-void setup_wifi()
-{
+void init_config() {
+  Memory memory;
+
+  // Load Config From Memory
+  config = memory.Load();
+
+  // Apply Overrides
+  config.applyOverrides();
+#ifndef DISABLE_WIFIMANAGER_SETUP
+  WifiManagerAdapter::setup(&config);
+#endif
+  pixel_server.config = &config;
+
+  channels = 3 * config.strip_length;
+  maxUniverses = channels / 512 + ((channels % 512) ? 1 : 0);
+  bool received[maxUniverses];
+  universesReceived = (bool *)malloc(sizeof(bool) * maxUniverses);
+  if (!universesReceived) { // failed malloc returns null
+    Serial.println(
```

Comment:
debugging mcus are such a pain sometimes... i wonder if theres another way to signify an error other than serial. like.... flashing first couple leds red or something? Youve done something similar in the past with other states